app/models/agents/http_status_agent.rb

```diff
@@ -84,11 +85,11 @@ def check_this_url(url, local_headers)
 
       # Deal with failures
       if measured_result.result
-        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.to_hash[:url]
+        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.env.url.to_s
         payload.merge!({ 'final_url' => final_url, 'redirected' => (url != final_url), 'response_received' => true, 'status' => current_status })
         # Deal with headers
         if local_headers.present?
-          header_results = measured_result.result.headers.select {|header, value| local_headers.include?(header)}
+          header_results = measured_result.result.headers.slice(*local_headers)
```

Comment:
Got it, I checked the [faraday docs](http://www.rubydoc.info/gems/faraday/Faraday%2FRequest%3Aheaders) which only say `headers` returns an`Object` which confused me. Do we need a spec to ensure it is kept case insensitive?